torch/_dynamo/variables/functions.py

```diff
@@ -573,6 +575,79 @@ def call_method(
                 assert tracer.symbolic_result is not None
                 return tracer.symbolic_result
+        elif name == "throw":
+            # * Raises an exception at the point where the generator was paused, and
+            # returns the next value yielded by the generator.
+            # * If the generator exits without yielding, raise StopIteration
+            # * If the generator function does not catch the passed-in exception,
+            # or raises a different exception, then that exception propagates to the caller.
+
+            if len(args) > 1:
+                raise IncorrectUsage(
+                    "the (type, exc, tb) signature of throw() is deprecated, "
+                    "use the single-arg signature instead."
+                )
+
+            # Setup the exception table and jump target in case of try...finally
+            tracer = self._get_inline_tracer(tx)
+            try:
+                self._setup_exception(tx, args[0])
+            except ObservedException:
+                # propagate the exception back to the parent caller
+                tx.exn_vt_stack.extend(tracer.exn_vt_stack)
+                raise
+
+            retval = self.next_variable(tx)
+
+            # We need to re-raise the exception one more time. But why? Let’s walk
+            # through an example and its generated bytecode: https://godbolt.org/z/ebdTbMv8M
+            #
+            #              ...
+            #         >>   58 PUSH_EXC_INFO
+            #
+            #   8          60 LOAD_GLOBAL              2 (ValueError)
+            #              70 CHECK_EXC_MATCH
+            #              72 POP_JUMP_IF_FALSE        7 (to 88)
+            #              74 POP_TOP
+            #
+            #   9          76 LOAD_CONST               3 (2)
+            #              78 YIELD_VALUE              3
+            #              80 RESUME                   1
+            #              82 POP_TOP
+            #              84 POP_EXCEPT
+            #              86 jump_backward           34 (to 20)
+            #              ...
+            #
+            # In this scenario, a generator can yield after `throw()` is called. Even
+            # after the exception is raised, it remains active within the `YIELD_VALUE`
+            # instruction. When the generator resumes, we cannot simply return to the
+            # `RESUME` point; instead, we must consult the exception table to determine
+            # the correct target.
+            #
+            # Re-raising the exception allows us to jump to the correct instruction
+            # given by the exception table. Without this step, if we let the trace
+            # proceed to the next instruction, it would follow the control flow where
+            # the exception raised by `throw()` was handled and swallowed, potentially
+            # leading to incorrect behavior.
+            self._setup_exception(tx, args[0])
```

Comment:
My problem with sending another exception in is, what if there's another except call around the generator? Does this trigger that except? Something like:

```
z = 0
def whoo():
    global z
    z = 0
    try:
	    try:
	        yield 1
	    except ValueError:
	        yield 2
	    finally:
	        z += 1
    except ValueError:
        z += 100
    z += 10
```

Reply:
This works as expected. Only the first ValueError + finally will be executed once one calls `.throw(valueError)`.

Reply:
> My problem with sending another exception in is, what if there's another except call around the generator? Does this trigger that except?

I said it wouldn't trigger another exception, but it does! As a workaround, one can raise an internal exception that user code wouldn't capture it.